### DISCUSSION TITLE
TVB-ukbb installation on MacOS

### DISCUSSION THREAD

**Discussion by michefabb** [McIntosh-Lab/tvb-ukbb#183] [Help]
Dear all, I have posted this also on the TVB Users Google forum at https://groups.google.com/g/tvb-users/c/7Rkj6x9Lqjs , I repost it here hoping a help!

Dear TVB experts,

it looks my MRI data do not include the fieldmap as required to use the EBRAINS TVB Pipeline https://wiki.ebrains.eu/bin/view/Collabs/tvb-pipeline/ .
Hence I am trying to use the TVB-ukbb pipeline available at https://github.com/McIntosh-Lab/tvb-ukbb/ , however I get a errors during installation. I am using a local machine with MacOS Monterey.

I followed the guide at https://github.com/McIntosh-Lab/tvb-ukbb/wiki/Installing-on-a-General-HPC-System :
- I installed Vagrant and Singularity as described at https://docs.sylabs.io/guides/3.2/user-guide/installation.html#mac
- when pulling the Singularity container "singularity pull library://noahfl/tvb/tvb-ukbb:latest" I get an error "FATAL:   While getting image info: error decoding image: invalid ObjectId in JSON: "noahfl/tvb/tvb-ukbb:sha256.840a4427e34241fee7033ecebdcb4215a1256c77280ddf21d8ffdba149006947"
- even trying the pull commands at https://cloud.sylabs.io/library/noahfl/tvb/tvb-ukbb or trying to click Download, it looks it gets stuck.

Similarly if I download the Installer at https://github.com/McIntosh-Lab/tvb-ukbb/releases/tag/1.3 and then run the script file install_ukbb_singularity.sh , I get the following error:
Solving environment: failed

ResolvePackageNotFound: 

  - noahfl::gradunwarp

I see that the Installer files include also the following, can they help me:
- install_ukbb_norepo.sh
- install_ukbb.sh

I am not using them since the Wiki of the TVB-ukbb and the corresponding paper are suggesting to use the Singularity container to simplify the installation and use.

### **UPDATE OF TODAY**
Dear all,

just an update, after manually downloading the .sif from https://cloud.sylabs.io/library/noahfl/tvb/tvb-ukbb and renaming to tvb-ukbb.sif , launching Vagrant and then executing as stated in the guide at https://github.com/McIntosh-Lab/tvb-ukbb/wiki/Installing-on-a-General-HPC-System the command singularity shell --nv -B /home /vagrant/tvb-ukbb.sif  (in the Virtual Machine created with Vagrant, the .sif file is located in the directory /vagrant), I get an error:
FATAL:   container creation failed: mount /proc/self/fd/5->/usr/local/var/singularity/mnt/session/rootfs error: can't mount image /proc/self/fd/5: failed to mount squashfs filesystem: input/output error

Can anybody help?

**Comment by pmahon3** [comment]
Hey @michefabb,

We are looking into this.

In the mean time: the pipeline is well tested on linux (this may be a mac support issue) so if it is blocking your work and you have access to a linux machine or can run a linux virtual machine on your mac try to proceed that way.

Will keep you updated.

**Reply by michefabb** [reply]
@pmahon3 
Dear Patrick,

I greatly appreciate your support! I have used Vagrant to launch Syngularity on a Linux VM on my Mac (Intel Silicon), but I received the error above…

Feel free to let me know if you need any further information from my side.

Thanks again and looking forward,

Michelangelo

**Reply by pmahon3** [reply]
Thanks Michelangelo. That's useful info

**Reply by michefabb** [reply]
@pmahon3
Dear All,

I would kindly ask you if you have any feedback to the above.
I did not mention explicitly that the Singularity installation website https://docs.sylabs.io/guides/3.2/user-guide/installation.html#mac mandates MacOS users to use Vagrant to launch Singularity containers, reason why I am using this latter as VM.

Hope it helps,

Miche

**Reply by pmahon3** [reply]
Hey @michefabb,

I'm currently working from a windows machine but will have a mac soon, unfortunately the order was delayed. Should have an update on this by the end of the month.

Patrick

**Reply by michefabb** [reply]
@pmahon3 thank you very much, really appreciated!

PS unfortunately it looks I cannot use the EBRAINS TVB-Image Processing Pipeline https://wiki.ebrains.eu/bin/view/Collabs/tvb-pipeline/ since my MRI data do not include the fieldmap as needed from this latter...would you have any suggestion / workaround in the meanwhile? :)
